src/System.Management.Automation/engine/NativeCommandProcessor.cs

```diff
@@ -662,11 +661,47 @@ private void InitNativeProcess()
                 }
                 else
                 {
+#if UNIX
                     _isRunningInBackground = true;
                     if (!startInfo.UseShellExecute)
                     {
                         _isRunningInBackground = isWindowsApplication;
                     }
+#else
+                    // Notice about Windows Store applications.
+                    //   If we run a process like Windows Terminal with wt.exe
+                    // the wt.exe process is replaced with WindowsTerminal.exe process
+                    // so wt.exe process is already terminated (HasExited is true)
+                    // and it makes no sense to wait the process - we consider it as a background process.
+                    //   If we run a process like Skype with skype.exe
+                    // the process is not terminated so we can get real path to the exe file:
+                    // - before run "c:\users\username\appdata\local\microsoft\windowsapps\skype.exe"
+                    // - after run the real path is "C:\Program Files\WindowsApps\Microsoft.SkypeApp_15.78.159.0_x86__kzf8qxf38zg5c\Skype\Skype.exe"
+                    // With the real path we can read the exe file and analyze its header.
+                    _isRunningInBackground = startInfo.UseShellExecute || _nativeProcess.HasExited;
+                    if (!_isRunningInBackground)
+                    {
+                        try
+                        {
+                            var mainModuleFileName = _nativeProcess.MainModule?.FileName;
+                            if (mainModuleFileName is not null && !string.Equals(startInfo.FileName, mainModuleFileName, StringComparison.OrdinalIgnoreCase))
+                            {
+                                using var exeStream = new FileStream(mainModuleFileName, FileMode.Open, FileAccess.Read, FileShare.Read);
```

Comment:
Can't you just use the `IsWindowsApplication()` method?

Reply:
We need the workaround just because `IsWindowsApplication()` today doen't work for AppX reparse points.

Reply:
I'm not saying relying on `isWindowsApplication` defined way above, but using `mainModuleFileName` passed to `IsWindowsApplication()`.  Isn't lines 689-690 doing essentially the same thing?

Reply:
Ah, yes, we can:-)

Reply:
Done.